### DISCUSSION TITLE
Can IceBridge be used to  implement p2p communication?

### DISCUSSION THREAD

**Discussion by playg8** [zeroc-ice/ice#2914] [General]
I want to implement p2p with iceBridge。 This can skip the server and save time。But I dont know how。https://doc.zeroc.com/ice/3.7/ice-services/icebridge. Can somebody give me some clue for this? Thank a lot.

**Comment by pepone** [comment]
When you use IceBridge, peers communicate through the bridge which is the contrary of p2p. In general with Ice you need at least one peer to have an public address where it listen for incoming connections. Ice doesn't directly support p2p communications where all parties are behind NATs.

**Reply by playg8** [reply]
Not very good, because bridge do not really support p2p. If there is one public address on one peer, It's not need p2p. 
Thank you peptone for answer my question!!!
